soh/src/code/z_parameter.c

```diff
@@ -2376,37 +2378,49 @@ u16 Randomizer_Item_Give(GlobalContext* globalCtx, GetItemEntry giEntry) {
             case RG_FOREST_TEMPLE_MAP:
             case RG_FOREST_TEMPLE_COMPASS:
             case RG_FOREST_TEMPLE_SMALL_KEY:
+            case RG_FOREST_TEMPLE_KEY_RING:
             case RG_FOREST_TEMPLE_BOSS_KEY:
                 mapIndex = SCENE_BMORI1;
+                maxKeys = FOREST_TEMPLE_SMALL_KEY_MAX;
```

Comment:
it's not quite clear how `maxKeys` is working here, but it seems this makes it so maxKeys is set when we get the map or compass or any key/keyring from a dungeon. is that intended?